content/addons.md

```diff
@@ -68,27 +68,35 @@ ember addon <addon-name> [options]
 
 The result is the creation of a directory called `<addon-name>`, which has many files and looks a bit like an Ember app. We won't need to use all the files to make a useful addon. By convention, _most_ Ember addons start with `ember` in the name, like `ember-basic-dropdown`. This will help other developers find our addon.
 
-<!-- Should we cover in-repo addons at all??? -->
-<!-- 
-If the addon is just meant to be used in a single project, an "in-repo" addon could be created instead. The benefit is that it is lightweight, but there are some major limitations; an in-repo addon can't be shared between apps, versioned independently, or published to npm. From within an existing Ember app, use:
+### Addon file structure
 
-```bash
-ember generate in-repo-addon <addon-name> [options]
-```
+In some ways, an addon is like a mini Ember app. It has a very similar file structure, uses a lot of the same API methods, and can do most things that components are able to do. 
 
-This generates a folder called `lib/<addon-name>` that contains its own `package.json` and an `index.js` file. 
--->
+Let's take a look a some of the most important files and folders in an addon, and how they are different from what we would find in an app. 
 
-### Addon file structure
+#### `addon/` 
 
-In some ways, an addon is like a mini Ember app. It has a very similar file structure, uses a lot of the same API methods, and can do most things that components are able to do. 
+If an addon only offers one UI template, the addon files usually go in the `addon/` directory. When the addon is used in a template, it is called by the "real name" of the addon, like `{{my-addon-name someValue=true}}`. This directory can hold many of the same subdirectories and files that an Ember app would, like `/components/` and `/templates/`.
+
+#### `app/` 
+
+If an addon will export multiple templates, the addon's files are referenced from the `app/` directory. When the addon templates are used, they are called by the template name, not the addon name. For example, the addon could provide `{{super-hero-img}}` and `{{fantastic-form}}`. Many addon developers prefix these kinds of templates with a nickname in order to avoid namespace clashes. For example, an addon template named `{{table-row}}` might cause problems if the app using it has a component by the same name, so `{{bs-table-row}}` is safer.
```

Comment:
Leading with "multiple templates" as the reason to use `/app` is misleading. The distinction between what `/app` and `/addon` do has nothing to do with whether there are multiple templates.

The distinction is all about namespacing. Here's an example:

1. Imagine an app named "the-app" that uses the "my-cool-addon" addon. 
2. Inside my-cool-addon there are files named `/addon/first.js` and `/app/second.js`.
3. For the app to import `first.js`, it must say `import something from "my-cool-addon/first"`.
4. For the app to import `second.js`, it must say `import something from "the-app/second"`. This is pretty weird! It makes it hard for somebody reading the code to tell that the file is really coming from `my-cool-addon`. 

So generally speaking, you want to always keep code that users of your addon are going to `import` inside your `/addon` folder, so it's properly namespaced under your addon's name.

So why does `/app` exist at all? Mostly because of how Ember discovers things like

 - components
 - helpers
 - services

For those things, _it only looks in the app's namespace_. That is, if the app says `{{some-thing message="hi"}}` in a template, Ember will effectively try each of these:
 - `import someComponent from "the-app/components/some-thing"`
 - `import someHelper from "the-app/helpers/some-thing"`

So the default way to make a component in an addon is to put the implementation in `/addon` (so that it's importable from the right namespace if somebody wants to extend it) but also reexport it from the `/app` namespace (so Ember can discover it automatically).

Now, this all might seem especially horrible, but it's mitigated by the fact that if you run `ember generate component` in an addon, it takes care of all this re-exporting business for you automatically as part of the blueprint, so addon authors for the most part can not think about it.

But the distinction is definitely annoying, and getting rid of it is one of the most important benefits of Module Unification. I'm not sure you want to bring that into this guide, but once it's widely deployed it will allow simplifying the guide.

Reply:
@jenweber basing on @ef4  comment  I suggest to change the section to something like this:

When resolving components, helpers and services Ember application uses `app` namespace. because of that, the default way to make a component in an addon is to put the implementation in `/addon` (so that it's importable from the right namespace if somebody wants to extend it) but also reexport it from the /app namespace (so Ember can discover it automatically).

Fortunately, when you run `ember g component foo-bar` in an addon, it takes care of all this re-exporting business for you automatically as part of the blueprint, so addon authors for the most part can not think about it.